session/session.go

```diff
@@ -2877,6 +2877,21 @@ func (s *session) GetTxnWriteThroughputSLI() *sli.TxnWriteThroughputSLI {
 	return &s.txn.writeSLI
 }
 
+var _ telemetry.TemporaryTableFeatureChecker = &session{}
+
+// TemporaryTableExists is used by the telemetry package to avoid circle dependency.
+func (s *session) TemporaryTableExists() bool {
+	is := domain.GetDomain(s).InfoSchema()
```

Comment:
I would recommend `s.GetInfoSchema()` if your intention also includes history/snapshot/transaction cases. It is introduced by me to get the correct infoschema by the context.

Reply:
Here it **should be** the latest schema, rather than a snapshot.